tst/EnergyPlus/unit/InternalHeatGains.unit.cc

```diff
@@ -2473,3 +2473,132 @@ TEST_F(EnergyPlusFixture, InternalHeatGains_WarnMissingInletNode)
+        "  Data Center Servers Power fLoadTemp,  !- Name",
+        "  -1.0,                    !- Coefficient1 Constant",
+        "  1.0,                     !- Coefficient2 x",
+        "  0.0,                     !- Coefficient3 x**2",
+        "  0.06667,                 !- Coefficient4 y",
+        "  0.0,                     !- Coefficient5 y**2",
+        "  0.0,                     !- Coefficient6 x*y",
+        "  0.0,                     !- Minimum Value of x",
+        "  1.5,                     !- Maximum Value of x",
+        "  -10,                     !- Minimum Value of y",
+        "  99.0,                    !- Maximum Value of y",
+        "  0.0,                     !- Minimum Curve Output",
+        "  99.0,                    !- Maximum Curve Output",
+        "  Dimensionless,           !- Input Unit Type for X",
+        "  Temperature,             !- Input Unit Type for Y",
+        "  Dimensionless;           !- Output Unit Type",
+
+        "  Curve:Biquadratic,",
+        "  Data Center Servers Airflow fLoadTemp,  !- Name",
+        "  -1.4,                    !- Coefficient1 Constant",
+        "  0.9,                     !- Coefficient2 x",
+        "  0.0,                     !- Coefficient3 x**2",
+        "  0.1,                     !- Coefficient4 y",
+        "  0.0,                     !- Coefficient5 y**2",
+        "  0.0,                     !- Coefficient6 x*y",
+        "  0.0,                     !- Minimum Value of x",
+        "  1.5,                     !- Maximum Value of x",
+        "  -10,                     !- Minimum Value of y",
+        "  99.0,                    !- Maximum Value of y",
+        "  0.0,                     !- Minimum Curve Output",
+        "  99.0,                    !- Maximum Curve Output",
+        "  Dimensionless,           !- Input Unit Type for X",
+        "  Temperature,             !- Input Unit Type for Y",
+        "  Dimensionless;           !- Output Unit Type",
+
+        " Curve:Quadratic,",
+        "  ECM FanPower fFlow,      !- Name",
+        "  0.0,                     !- Coefficient1 Constant",
+        "  1.0,                     !- Coefficient2 x",
+        "  0.0,                     !- Coefficient3 x**2",
+        "  0.0,                     !- Minimum Value of x",
+        "  99.0;                    !- Maximum Value of x",
+    });
+
+    ASSERT_TRUE(process_idf(idf_objects));
+
+    bool ErrorsFound(false);
+    state->dataGlobal->NumOfTimeStepInHour = 1;
+    state->dataGlobal->MinutesPerTimeStep = 60;
+
+    HeatBalanceManager::GetZoneData(*state, ErrorsFound);
+    ASSERT_FALSE(ErrorsFound);
+    state->dataHeatBalFanSys->MAT.allocate(1);
+    state->dataHeatBalFanSys->ZoneAirHumRat.allocate(1);
+
+    state->dataHeatBalFanSys->MAT(1) = 24.0;
+    state->dataHeatBalFanSys->ZoneAirHumRat(1) = 0.008;
+
+    InternalHeatGains::GetInternalHeatGainsInput(*state);
+    ASSERT_FALSE(ErrorsFound);
+
+    state->dataGlobal->numSpaces = 0;
```

Comment:
One would not think to add this line, numSpaces=0, since it initializes to 0, so out of curiosity, did you need to add this line for some reason?

Reply:
@rraustad There was no specific reason.  I was being very "definitive" just to make sure that I had control over that variable.  I can try removing it--if it's initialized properly then there should be no change in how the test runs.

Reply:
I wasn't suggesting a change but I suspect you will save the zoneEqIndex (or zoneEqConfigIndex?) to speed things up. You can delete this line for that commit.

Reply:
Confirming here that getting rid of that line does not cause any problems.  So, this removal will get updated in the next commit.